contracts/mocks/Create2Impl.sol

```diff
@@ -4,13 +4,13 @@ import "../utils/Create2.sol";
 import "../token/ERC20/ERC20.sol";
 
 contract Create2Impl {
-    function deploy(bytes32 salt, bytes memory code) public {
-        Create2.deploy(salt, code);
+    function deploy(uint256 value, bytes32 salt, bytes memory code) public {
+        Create2.deploy(value, salt, code);
```

Comment:
Should I make this function payable and send the ether here?

Reply:
I _think_ you won't be able to do this because the contract won't yet have the received balance credited. I'd add a `receive` function, send funds to the contract, and then call `deploy`:

```solidity
receive() payable external {}
```

Reply:
@nventuro ty ty I was having trouble figure this out didn't know if it should only reside in the mock etc.